### ISSUE TITLE
About implementation of KL divergence Loss

### ISSUE THREAD

**Issue by Rheelt** [yabufarha/ms-tcn#2]
Thank you for your contribution! I use torch.nn.KLDivLoss to implement KL Loss, but the loss I get is NAN. So can you tell me the details of KL Loss implemented in MS-TCN.
Thanks.

**Comment by yabufarha** [comment]
Hi, 
We used the following line of code for the KL divergence loss:
self.kldiv(F.log_softmax(p[:, :, 1:], dim=1), F.softmax(p.detach()[:, :, :-1], dim=1))
where 
self.kldiv = nn.KLDivLoss()
As in the T-MSE loss, This loss is also weighted with \lambda = 0.15

I hope this would help.

Best,
Yazan

**Comment by Rheelt** [comment]
Thanks for your reply.
